docs-gh-pages/checkpoint-block-validation.md

```diff
@@ -0,0 +1,42 @@
+# CheckpointBlock Validation
+
+Each transaction sent to cluster needs to be validated before proposing it to block/consensus.
+Part of code responsible for validation is in `TransactionValidator.scala`.
+
+## Duplicated transaction validation
+Duplicated transaction can affect wallet balances so block which contains transaction with specific `hash` twice
+is incorrect and needs to be rejected. Method `validateDuplicatedTransactions` returns `DuplicatedTransaction`
+validation result if there are any duplicated transactions.
+
+![alt text](img/checkpoint-block-validation/duplicated-transaction.jpg "Duplicated transaction")
+
+## Signature validation and hash integrity
+Each block has its unique signature which is created by signing block by private key.
+Thanks to that we know that block was created for sure by known signer.
+Each block has also unique `hash` which is generated from the content of block so if someone malformed the content
+of block - the hash should be different. If hash is different or does not match the content of block then such block
+is rejected because is malformed.
+
+## Balance validation
```

Comment:
I think we should add info about double spend validation somewhere and a hint about blacklisting. If address performed a double spend it's actually being blacklisted forever (I think until rollback actually but Wyatt mentioned it was intended to be forever).

Reply:
Though I'm not sure if double spend can be even performed at this point with checks we introduced in POST /transaction endpoint and pull for consensus logic. Still the check is there.

Reply:
But maybe that's not a part of checkpoint block validation, but rather docs for checkpointAcceptanceService